Storage/ThirdParty/SwiftData.swift

```diff
@@ -65,6 +67,21 @@ public class SwiftData {
         assert(sqlite3_threadsafe() == 2)
     }
 
+    public func interrupt() {
+        var connection: SQLiteDBConnection?
+
+        var error: NSError? = nil
+        if SwiftData.ReuseConnections {
+            connection = getSharedConnection(&error)
+        } else {
+            assertionFailure("Interrupts don't make sense without a shared connection")
```

Comment:
Lemme make sure I understand this.

Interrupts don't make sense without a shared connection because we're not keeping a reference to the connection in the Deferred, right?

So isn't another way to do this to have *two* `interrupt` methods, one of which takes a connection argument? That way we can move towards the best possible world, where we keep the connection in the `Deferred`, and can thus cancel it directly?